static/app/views/performance/utils/useModuleURL.tsx

```diff
@@ -0,0 +1,55 @@
+import partial from 'lodash/partial';
+
+import useOrganization from 'sentry/utils/useOrganization';
+import {normalizeUrl} from 'sentry/utils/withDomainRequired';
+import {BASE_URL as AI_BASE_URL} from 'sentry/views/aiMonitoring/settings';
+import {BASE_URL as RESOURCES_BASE_URL} from 'sentry/views/performance/browser/resources/settings';
+import {BASE_URL as VITALS_BASE_URL} from 'sentry/views/performance/browser/webVitals/settings';
+import {BASE_URL as CACHE_BASE_URL} from 'sentry/views/performance/cache/settings';
+import {BASE_URL as DB_BASE_URL} from 'sentry/views/performance/database/settings';
+import {BASE_URL as HTTP_BASE_URL} from 'sentry/views/performance/http/settings';
+import {BASE_URL as APP_STARTS_BASE_URL} from 'sentry/views/performance/mobile/appStarts/settings';
+import {BASE_URL as SCREEN_LOADS_BASE_URL} from 'sentry/views/performance/mobile/screenload/settings';
+import {BASE_URL as MOBILE_UI_BASE_URL} from 'sentry/views/performance/mobile/ui/settings';
+import {BASE_URL as QUEUE_BASE_URL} from 'sentry/views/performance/queues/settings';
+import {INSIGHTS_BASE_URL} from 'sentry/views/performance/settings';
+import {ModuleName} from 'sentry/views/starfish/types';
+
+const MODULE_BASE_URLS: Record<ModuleName, string> = {
+  [ModuleName.DB]: DB_BASE_URL,
+  [ModuleName.HTTP]: HTTP_BASE_URL,
+  [ModuleName.CACHE]: CACHE_BASE_URL,
+  [ModuleName.QUEUE]: QUEUE_BASE_URL,
+  [ModuleName.SCREEN_LOAD]: SCREEN_LOADS_BASE_URL,
+  [ModuleName.APP_START]: APP_STARTS_BASE_URL,
+  [ModuleName.VITAL]: VITALS_BASE_URL,
+  [ModuleName.RESOURCE]: RESOURCES_BASE_URL,
+  [ModuleName.AI]: AI_BASE_URL,
+  [ModuleName.MOBILE_UI]: MOBILE_UI_BASE_URL,
+  [ModuleName.OTHER]: '',
+  [ModuleName.ALL]: '',
+};
+
+export const useModuleURL = (moduleName: ModuleName): string => {
+  const {slug} = useOrganization();
+
+  if (moduleName === ModuleName.AI) {
+    // AI Doesn't live under "/performance"
+    return normalizeUrl(`/organizations/${slug}${AI_BASE_URL}`);
+  }
+
+  return normalizeUrl(
+    `/organizations/${slug}${INSIGHTS_BASE_URL}/${MODULE_BASE_URLS[moduleName]}`
+  );
+};
+
+export const useDatabaseModuleURL = partial(useModuleURL, ModuleName.DB);
```

Comment:
From discoverability POV, not everyone will know the names of these modules which will make it less likely for them to reach for their usage as the different variations will be fragmented around the codebase and make it more likely that they will hardcode the URLs like they are used to today (or add new module definitions)

Have you thought about changing the hook signature to accept the module name as a string? I think that would help with eliminating the need to import the name enum while preserving type safety and also improving the typeahead experience when using the hook.

From a DX pov, the use of enums can sometimes be overly restrictive, which I suspect is part of why these partial types even exist here in the first place.

Reply:
@JonasBa :eyes:

> From discoverability POV, not everyone will know the names of these modules which will make it less likely for them to reach for their usage as the different variations will be fragmented around the codebase and make it more likely that they will hardcode the URLs like they are used to today (or add new module definitions)

I don't follow! What do you mean by "the different variations will be fragmented"? Also, how does not knowing the module names result in more people hardcoding the URLs or adding new module definitions?

Do you mean that people might add a new module but not add the partial URL function? That's fair if yes. Or that people will use the partials sometimes, and `useModuleURL` other times? Is that bad?

Also, I'd _love_ to fully prevent hard-coded URLs, but I don't know of a lightweight way of doing that!

> Have you thought about changing the hook signature to accept the module name as a string? I think that would help with eliminating the need to import the name enum while preserving type safety and also improving the typeahead experience when using the hook.

That sounds reasonable! I think changing `ModuleName` to a string union type instead of an enum is probably a good idea overall. I'm happy to do that in a follow-up.

> From a DX pov, the use of enums can sometimes be overly restrictive, which I suspect is part of why these partial types even exist here in the first place.

It is not. The partial functions exist because for one thing, that's the API I prefer _and_ because there was already an existing partial hook (`useCacheUrl`) so I set up all partials for consistency. I don't particularly love the partials, but they seem better than always importing `useModuleURL`. I'm fine with either though, if you really don't like the partial!